ical/parsing/parser.py

```diff
@@ -25,96 +25,133 @@
+    line_len = len(line)
+    pos = 0
+
+    # parse NAME
+    while True:
+        if pos >= line_len:
+            raise CalendarParseError(f"Unexpected end of line. Expected ';' or ':'", detailed_error=line)
+        if line[pos] in (';', ':'):
+            name = line[0:pos]
+            if not _RE_NAME.fullmatch(name):
+                raise CalendarParseError(f"Invalid property name '{name}'", detailed_error=line)
+            result.name = name.lower()
+            break
+        pos += 1
+
+    # parse PARAMS if any
+    if line[pos] == ';':
+        result.params = []
+        pos += 1
+        params_start = pos
+        all_params_read = False
+
+        while pos < line_len and not all_params_read:
+            
+            # Read until we hit name/value separator (=)
+            if line[pos] != '=':
+                pos += 1
+                if pos >= line_len:
+                    raise CalendarParseError(f"Unexpected end of line. Expected '='", detailed_error=line)
+                continue
+                
+            # param name reached
+            param_name = line[params_start:pos]
+            if not _RE_NAME.fullmatch(param_name):
+                raise CalendarParseError(f"Invalid parameter name '{param_name}'", detailed_error=line)
+            pos += 1
+            
+            # Now read values. (list separated by comma)
+            param_values = []
+            param_value_start = 0
+            all_values_read = False
+            
+            while not all_values_read:
+                param_value_start = pos
+                quoted = False
+
+                if line[pos] == '"':
+                    # parameter value is quoted
+                    quoted = True
+                    pos += 1
+
+                param_value_read = False
+                while not param_value_read:
+                    if pos >= line_len:
+                        if quoted:
+                            raise CalendarParseError(f"Unexpected end of line. Expected end of qouted string", detailed_error=line)
+                        else:
+                            raise CalendarParseError(f"Unexpected end of line. Expected ',', ';' or ':'", detailed_error=line)
+                    
+                    char = line[pos]
+
+                    if char == '"':
+                        if not quoted:
+                            raise CalendarParseError(f"Unexpected quote character outside parameter value.", detailed_error=line)
+                        param_value = line[param_value_start + 1:pos]
+                        param_value_read = True
+                        pos += 1
+                        char = line[pos]
+                    elif not quoted and char in (',', ';', ':'):
+                        param_value = line[param_value_start:pos]
+                        param_value_read = True
+
+                    if param_value_read:
+                        if not char in (',', ';', ':'):
```

Comment:
I think this only applies when exiting the quoting so move this inside that branch?